src/headers.rs

```diff
@@ -0,0 +1,179 @@
+//! This module provides access to the HTTP MIME headers
+//! of a request or response.
+//!
+//! Not to be confused with the request/response metadata
+//! which is also called "headers" in TrafficServer
+
+use crate::bindings::*;
+use crate::helpers::checked_raw_str;
+use std::marker::PhantomData;
+
+/// An iterator over header field values
+///
+/// Skips any invalid strings
+pub struct HeaderValueIter<'a, 'b, T: 'a> {
+    field: &'b HeaderField<'a, 'b, T>,
+    index: usize,
+    length: usize,
+}
+
+impl<'a, 'b, T> Iterator for HeaderValueIter<'a, 'b, T> {
+    type Item = &'a str;
+    fn next(&mut self) -> Option<Self::Item> {
+        while self.index < self.length {
+            let res = checked_raw_str(|len| unsafe {
+                TSMimeHdrFieldValueStringGet(
+                    self.field.headers.bufp,
+                    self.field.headers.offset,
+                    self.field.raw_field,
+                    self.index as i32,
+                    len,
+                )
+            });
+            self.index += 1;
+            if let Ok(res) = res {
+                return Some(res);
+            }
+        }
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (0, Some(self.length))
+    }
+}
+
+/// A single MIME header field
+///
+/// Has a name and contains multiple values.
+pub struct HeaderField<'a, 'b, T: 'a> {
+    headers: &'b Headers<'a, T>,
+    raw_field: TSMLoc,
+}
+
+impl<'a, 'b, T> Drop for HeaderField<'a, 'b, T> {
+    fn drop(&mut self) {
+        unsafe { TSHandleMLocRelease(self.headers.bufp, self.headers.offset, self.raw_field) };
+    }
+}
+
+impl<'a, 'b, T> HeaderField<'a, 'b, T> {
+    /// Returns the name of this field
+    pub fn get_name(&self) -> Result<&str, crate::Error> {
+        let name = checked_raw_str(|len| unsafe {
+            TSMimeHdrFieldNameGet(self.headers.bufp, self.headers.offset, self.raw_field, len)
+        })?;
+        Ok(name)
+    }
+
+    /// Returns an Iterator over all values for this field
+    ///
+    /// Any invalid strings will be skipped.
+    pub fn get_value_iter(&'b self) -> HeaderValueIter<'a, 'b, T> {
+        let length = unsafe {
```

Comment:
`TSMimeHdrFieldValuesCount` returns `int`. Maybe check the value from TS is >= 0?